src/strands/event_loop/event_loop.py

```diff
@@ -264,8 +278,41 @@ async def event_loop_cycle(agent: "Agent", invocation_state: dict[str, Any]) ->
         yield ForceStopEvent(reason=e)
         logger.exception("cycle failed")
         raise EventLoopException(e, invocation_state["request_state"]) from e
+    finally:
+        if output_schema:
+            handler = StructuredOutputHandler(output_schema)
+            handler.cleanup_all_state(invocation_state)
+
+    # Force structured output tool call if LLM didn't use it automatically
+    if output_schema and stop_reason != "tool_use":
+        invocation_state.setdefault("_structured_output_attempts", 0)
+        if invocation_state["_structured_output_attempts"] >= _MAX_STRUCTURED_OUTPUT_ATTEMPTS:
+            logger.warning(
+                f"Structured output forcing exceeded maximum attempts ({_MAX_STRUCTURED_OUTPUT_ATTEMPTS}), returning without structured output"
+            )
+            yield EventLoopStopEvent(
+                stop_reason, message, agent.event_loop_metrics, invocation_state["request_state"], None
+            )
+            return
+
+        invocation_state["_structured_output_attempts"] += 1
+        logger.debug(
+            f"Forcing structured output tool, attempt {invocation_state['_structured_output_attempts']}/{_MAX_STRUCTURED_OUTPUT_ATTEMPTS}"
+        )
+        agent.messages.append({
+            "role": "user", 
+            "content": [{"text": "You must format the previous response as structured output."}]
```

Comment:
We're being a bit opinated here on the message - what happens if they're in another language?  How can we let people customize this?